pkg/controller/webhooks/deletionprotection.go

```diff
@@ -145,3 +151,40 @@ func isDeletionProtected(obj client.Object) bool {
 	// If the value is "true" the protection is disabled.
 	return !(val == "true")
 }
+
+func getControlPlaneClient(mgr ctrl.Manager) (client.Client, error) {
+
+	cpClient := mgr.GetClient()
+	if viper.IsSet("CONTROL_PLANE_KUBECONFIG") {
+		kubeconfigPath := viper.GetString("CONTROL_PLANE_KUBECONFIG")
+		file, err := os.Open(kubeconfigPath)
+		if err != nil {
+			return cpClient, err
+		}
+		defer file.Close()
+
+		// Get the file size
+		stat, err := file.Stat()
+		if err != nil {
+			return cpClient, err
+		}
+
+		// Read the file into a byte slice
+		kubeconfig := make([]byte, stat.Size())
+		_, err = bufio.NewReader(file).Read(kubeconfig)
+		if err != nil && err != io.EOF {
+			return cpClient, err
+		}
```

Comment:
even though I like C way of reading a file, I'd go with os.ReadFile() :)

Reply:
Good point. Changed the code to use `os.ReadFile()`